src/Service/HealthCheck/HealthCheckHelper.cs

```diff
@@ -57,6 +61,38 @@ public ComprehensiveHealthCheckReport GetHealthCheckResponse(HttpContext context
             return ComprehensiveHealthCheckReport;
         }
 
+        // Updates the incoming role header with the appropriate value from the request headers.
+        public void UpdateIncomingRoleHeader(HttpContext httpContext)
+        {
+            StringValues clientRoleHeader = httpContext.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
+            StringValues clientTokenHeader = httpContext.Request.Headers[AuthenticationOptions.CLIENT_PRINCIPAL_HEADER];
+
+            if (clientRoleHeader.Count == 1)
+            {
+                _incomingRoleHeader = clientRoleHeader.ToString().ToLowerInvariant();
+                _incomingRoleToken = clientTokenHeader.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Checks if the incoming request is allowed to access the health check endpoint.
+        /// Anonymous requests are only allowed in Development Mode.
+        /// </summary>
+        /// <param name="httpContext">HttpContext to get the headers.</param>
+        /// <param name="hostMode">Compare with the HostMode of DAB</param>
+        /// <param name="allowedRoles">AllowedRoles in the Runtime.Health config</param>
+        /// <returns></returns>
+        public bool IsUserAllowedToAccessHealthCheck(HttpContext httpContext, bool isDevelopmentMode, List<string> allowedRoles)
+        {
+            if (allowedRoles == null || allowedRoles.Count == 0)
+            {
+                // When allowedRoles is null or empty, all roles are allowed if Mode = Development.
+                return isDevelopmentMode;
+            }
+
+            return allowedRoles.Contains(_incomingRoleHeader);
```

Comment:
If we just want to check for presence of this `roleHeader`, shouldn't we create a `HashSet` for allowedRoles instead of a List<string> so that its O(1) time complexity?